### ISSUE TITLE
Zowe 1.0.0 install script regression with proclib detection

### ISSUE THREAD

**Issue by kiril-ibm** [zowe/zowe-install-packaging#304]
On the same zOS instance I am able to install Zowe 0.9.5 and it finds CAN.USER.PROCLIB to deploy ZOWESVR.jcl

Console output:
Attempting to create ZOWESVR PROCLIB member ...
FSUM6764 logname: cannot get login name
  PROC ZOWESVR placed in CAN.USER.PROCLIB

It fails to do the same for Zowe ​1.0.0-20190124.170120-449 

Console output:
Attempting to create ZOWESVR PROCLIB member ...
FSUM6764 logname: cannot get login name
rm: FSUM9195 cannot unlink entry "mstjcl00": EDC5129I No such file or directory.
Unable to create the PROCLIB member 
  Failed to put ZOWESVR.JCL in a PROCLIB dataset.
  Please add it manually from /var/zowe/zowe/zowe-1.0.0/ZOWESVR.JCL to your PROCLIB
    To find PROCLIB datasets, issue /$D PROCLIB in SDSF

**Comment by kiril-ibm** [comment]
[2019-01-29-11-44-29.log](https://github.com/zowe/zowe-install-packaging/files/2808730/2019-01-29-11-44-29.log)

**Comment by jordanCain** [comment]
relevant section of log: 
```<zowe-copy-proc.sh>
LOGNAME=CLJIA
USER=CLJIA
MAIL=/usr/mail/CLJIA
HOME=/u/cljia
whoami=CLJIA
logname=
userid obtained from LISTUSER is CLJIA
userid=[CLJIA]
Old dataset CLJIA.zowetemp.clist did not exist
Member name ZOWESVR was specified
Member name ZOWESVR was used
'auto' was requested: choose PROCLIB dataset automatically
Try JES2 PROCLIB concatenation
  Unable to read JES2 PROCLIB concatenation with opercmd REXX exec
Try master JES2 JCL
Unable to read master JES2 JCL
Try SYS1.PROCLIB
Unable to write to SYS1.PROCLIB
Unable to create the PROCLIB member 
</zowe-copy-proc.sh>```

I think we would need to also see the log file of 0.9.5, @John-A-Davies what do you think?

**Comment by kiril-ibm** [comment]
[2019-01-30-17-24-25.log](https://github.com/zowe/zowe-install-packaging/files/2815475/2019-01-30-17-24-25.log)
0.9.5 log file

**Comment by John-A-Davies** [comment]
The message `Unable to read JES2 PROCLIB concatenation with opercmd REXX exec` indicates that `opercmd` failed to work at all.  Probable cause is that `opercmd` is not present in `/var/zowe/zowe/zowe-0.9.5/scripts` or is not readable and executable by the **installing user**.

**Comment by John-A-Davies** [comment]
I think there was a problem with opercmd prior to zowe-1.0.0.  The build did not set permission `-rwxr-xr-x` on the opercmd file (it only did this for `*.sh` files), so opercmd was not executable when un-PAXed into `$INSTALL_DIR/scripts`.  This is fixed in 1.0.0.  

The PROCLIB member is created by `zowe-copy-proc.sh` which depends on opercmd for it to work.  Your message about `opercmd REXX exec` means opercmd did not work.

**Comment by stevenhorsman** [comment]
@kiril-ibm - can you provide the ls listing that John requested (I think it will be `ls -al <install_dir>/scripts/` )

**Comment by kiril-ibm** [comment]
CLJIA:/SP14/var/zowe/zowe/zowe-0.9.5/scripts #>pwd
/SP14/var/zowe/zowe/zowe-0.9.5/scripts
CLJIA:/SP14/var/zowe/zowe/zowe-0.9.5/scripts #>ls -la
total 176
drwxrwxrwx   3 CLJIA    #IBMTSPT    8192 Jan 30 17:25 .
drwxr-xr-x  30 CLJIA    #IBMTSPT    8192 Jan 31 04:44 ..
drwxr-xr-x   2 CLJIA    #IBMTSPT    8192 Jan 30 17:25 internal
-rwxrwxrwx   1 CLJIA    #IBMTSPT    3371 Jan 30 17:25 zowe-runtime-authorize.sh
-rwxrwxrwx   1 CLJIA    #IBMTSPT    1069 Jan 30 17:25 zowe-start.sh
-rwxrwxrwx   1 CLJIA    #IBMTSPT    1024 Jan 30 17:25 zowe-stop.sh
-rwxrwxrwx   1 CLJIA    #IBMTSPT   35112 Jan 30 17:25 zowe-verify.sh

**Comment by stevenhorsman** [comment]
Sorry @kiril-ibm - I meant the /scripts folder in the install directory (where the pax gets expanded), not the runtime directory, so we can see the opercmd permissions. It would be interesting to compare and constrast 1.0 vs 0.9.5 as John points out we've made some changes which should have made things better, not worse

**Comment by kiril-ibm** [comment]
@stevenhorsman all information has been provided directly to John on the slack channel

**Comment by kiril-ibm** [comment]
CLJIA:/SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/scripts #>pwd
/SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/scripts
CLJIA:/SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/scripts #>ls -la
total 528
drwxr-xr-x   2 CLJIA    #IBMTSPT    8192 Jan 29 15:01 .
drwxr-xr-x   7 CLJIA    #IBMTSPT    8192 Feb  1 08:47 ..
-rwxr-xr-x   1 CLJIA    #IBMTSPT     619 Jan 29 15:01 ocopyshr.clist
-rwxr-xr-x   1 CLJIA    #IBMTSPT     920 Jan 29 15:01 ocopyshr.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT     764 Jan 29 15:01 opercmd
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1445 Jan 29 15:01 run-zowe.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1267 Jan 29 15:01 zlux-install-script.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    9211 Jan 29 15:01 zowe-api-mediation-configure.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1396 Jan 29 15:01 zowe-api-mediation-install.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    4826 Jan 29 15:01 zowe-config-stc.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    7563 Jan 29 15:01 zowe-copy-proc.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    2572 Jan 29 15:01 zowe-explorer-api-configure.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1887 Jan 29 15:01 zowe-explorer-api-install.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    4479 Jan 29 15:01 zowe-explorer-ui-configure.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1757 Jan 29 15:01 zowe-explorer-ui-install.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    9136 Jan 29 15:01 zowe-init.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1420 Jan 29 15:01 zowe-install-existing-plugin.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    3121 Jan 29 15:01 zowe-install-iframe-plugin.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT   13931 Jan 29 15:01 zowe-parse-yaml.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT     643 Jan 29 15:01 zowe-prepare-runtime.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1634 Jan 29 15:01 zowe-runtime-authorize.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT     870 Jan 29 15:01 zowe-start.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT     871 Jan 29 15:01 zowe-stop.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT   37404 Jan 29 15:01 zowe-verify.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    1600 Jan 29 15:01 zowe-zlux-configure-certificates.sh
-rwxr-xr-x   1 CLJIA    #IBMTSPT    2777 Jan 29 15:01 zowe-zlux-configure-ports.sh
CLJIA:/SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/scripts

CLJIA:/SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0 #>ls -la
total 128
drwxr-xr-x   7 CLJIA    #IBMTSPT    8192 Feb  1 08:47 .
drwxr-xr-x   3 CLJIA    #IBMTSPT    8192 Feb  1 08:46 ..
drwxr-xr-x   5 CLJIA    #IBMTSPT    8192 Jan 29 15:02 files
drwxr-xr-x   2 CLJIA    #IBMTSPT    8192 Feb  1 08:46 install
drwxr-xr-x   2 CLJIA    #IBMTSPT    8192 Jan 29 15:01 licenses
drwxrwxrwx   2 CLJIA    #IBMTSPT    8192 Feb  1 08:46 log
-rw-r--r--   1 CLJIA    #IBMTSPT     280 Jan 29 15:01 manifest.json
drwxr-xr-x   2 CLJIA    #IBMTSPT    8192 Jan 29 15:01 scripts


John Davies [9:22 AM]
please check that all directory permissions down to and including opercmd (`/SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/scripts/opercmd`) have execute permission for all users.  Try running `zowe-check-prereqs.sh` first, because it should do that for you.


KIRIL STRELTSOV [9:22 AM]
ok


John Davies [10:56 AM]
any progress?


KIRIL STRELTSOV [10:56 AM]
Just came to work, executing

zowe-check-prereqs.sh what is location of the script ?

Script zowe-check-prereqs.sh started
Info: INSTALL_DIR environment variable is empty
Info: INSTALL_DIR environment variable is now set to /SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/install

Check entries in the install directory
OK

Check user has the basic auth to list the resources

Check RACF or other SAF security settings are correct
Error: user IZUSVR is not in group IZUADMIN

Check IZUGUEST
Error: user IZUGUEST is not in group IZUUNGRP

Is opercmd available?

OK: opercmd is available

Check z/OS RELEASE
release of z/OS is RELEASE z/OS 02.03.00 LICENSE = z/OS
OK: version 02.03.00 is supported

Check install user is a member of the IZUADMIN group
Installer’s TSO userid:
CLJIA
installer’s group(s):
IZUADMIN
OK

Check user can issue START/STOP commands
OK

Check user can issue SDSF commands
OK

Check the ports in the yaml file are not already in use
Error: port 27552 is already in use
Error: port 27553 is already in use
Error: port 27554 is already in use
Error: port 28545 is already in use
Error: port 28547 is already in use
Error: port 8542 is already in use

Check z/OSMF servers are up
OK

Check enough free space is available in target z/OS USS HFS install folder
Info: Existing directory to be checked is /var/zowe/zowe/zowe-1.0.0
OK

Check Node version
node: ./zowe-check-prereqs.sh 427: FSUM7351 not found
Error: can not determine node version

Check Java version
OK

Check CEE_RUNOPTS


Check USS AUTOCVT
OK: OMVS AUTOCVT is set to OFF
OK: _BPXK_AUTOCVT is not set
OK: _BPXK_CCSIDS is not set

Script zowe-check-prereqs.sh ended

**Comment by John-A-Davies** [comment]
Note that this test of 1.0.0 install is being run by Jenkins java, and some other commands such as `netstat` don't work either.  

The `opercmd` is a REXX EXEC, so it may be executed differently in this Jenkins java environment.

Also, this scenario fails only if you specify `dsName=auto` in the zowe-install.yaml file.

**Comment by kiril-ibm** [comment]
In our environment when we call your install scripts from java calling
**opercmd "d t"**
returns
opercmd: /SP14/var/zowe/jenkins/dl/zowe-1.0.0-20190129.200403-472/zowe-1.0.0/install/..//scripts/zowe-copy-proc.sh 194: FSUM7351 not found
To fix that you have to call it with ./
**./opercmd "d t"**
SP14      2019032  11:01:45.23             ISF031I CONSOLE MYCONS ACTIVATED
SP14      2019032  11:01:45.23            -d t
SP14      2019032  11:01:45.23  STC10147   IEE136I LOCAL: TIME=11.01.45 DATE=2019.032  UTC: TIME=19.01.45 DATE=2019.032
ISF754I Command 'SET CONSOLE MYCONS' generated from associated variable ISFCONS.
ISF776I Processing started for action 1 of 1.
ISF769I System command issued, command text: d t.
ISF766I Request completed, status: COMMAND ISSUED.

**Comment by John-A-Davies** [comment]
OK so you don't have dot in your path; I should fix this to cover your situation, which is common.

**Comment by John-A-Davies** [comment]
I am testing a fix to `zowe-copy-proc.sh`.

**Comment by John-A-Davies** [comment]
Fixed by #347
